genai-perf/genai_perf/parser.py

```diff
@@ -434,26 +439,42 @@ def _add_input_args(parser):
         "Alternatively, a string representing a json formatted dict can be provided.",
     )
 
+    input_group.add_argument(
+        "--goodput",
+        "-g",
+        nargs="+",
+        required=False,
+        help="An option to provide constraints in order to compute goodput. "
+        "Specify goodput constraints as 'key:value' pairs, where the key is a "
+        "valid metric name, and the value is a number representing "
+        "either milliseconds or a throughput value per second. For example, "
+        "'request_latency:300' or 'output_token_throughput_per_request:600'. "
+        "Multiple key:value pairs can be provided, separated by spaces. ",
+    )
+
     input_group.add_argument(
         "--input-file",
         type=file_or_directory,
         default=None,
         required=False,
         help="The input file or directory containing the content to use for "
-        "profiling. To use synthetic files for a converter that needs "
-        "multiple files, prefix the path with 'synthetic:', followed by a "
-        "comma-separated list of filenames. The synthetic filenames should "
-        "not have extensions. For example, 'synthetic:queries,passages'. "
-        "Each line should be a JSON object with a 'text' or 'image' field "
-        'in JSONL format. Example: {"text": "Your prompt here"}',
+        "profiling. Each line should be a JSON object with a 'text' or "
+        "'image' field 'in JSONL format. Example: {\"text\":"
+        ' "Your prompt here"}\'. To use synthetic files for a converter that '
+        "needs multiple files, prefix the path with 'synthetic:', followed "
+        "by a comma-separated list of filenames. The synthetic filenames "
+        "should not have extensions. For example, "
+        "'synthetic:queries,passages'. ",
     )
 
     input_group.add_argument(
+        "--num-payloads",
```

Comment:
Thoughts on `--num-dataset-entries`?

Reply:
I'd prefer the first, since this is used for the input file approach too.

@debermudez @nv-hwoo, any thoughts on the best name for this arg?

Reply:
Actually, do we want to use this for both synthetic and input file approaches? When the option is used for synthetic, it sets the total number of input data being generated, whereas in input file approach, it defines how many data we want to use/sample from the given input file. Feels like we need a separate option for each.. (although we are trying to cut-down the number of options 😂 )

Anyways to your question, I think I like
- `--input-data-size`
- `--input-sample-size`

Reply:
@nv-hwoo I actually think it makes sense to only have one option. They're both functionally doing the same thing. Picking how many dataset entries to generate/use for the purposes of what each request's inputs are, round-robin style.

Reply:
NooOOoOoOOooOoo!! I AM RIGHT 😤 
Yeah they're functionally the same, I agree.

Reply:
Thank you both for the feedback! I addressed all your comments and updated the code. It is currently running through CI.